### PR TITLE
Specialization setup with repository

### PR DESCRIPTION
**Pull request by TRybina132**

### CODE REVIEW

**Review by YuraRov — CHANGES_REQUESTED** [review]
Please name all async methods with suffix Async, for example, AddSpecializationAsync()

File: Core/Interfaces/Repositories/ISpecializationRepository.cs [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Core.Interfaces.Repositories
+{
+    public interface ISpecializationRepository
+    {
+        Task<ICollection<Specialization>> GetAllSpecializationsAsync();
```
Comment (YuraRov) [inline Core/Interfaces/Repositories/ISpecializationRepository.cs:12]:
```suggestion
        Task<IEnumerable<Specialization>> GetAllSpecializationsAsync();
```

File: DataAccess/Repositories/SpecializationRepository.cs [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+namespace DataAccess.Repositories
+{
+    public class SpecializationRepository : ISpecializationRepository
+    {
+        ClinicContext _context;
+        public SpecializationRepository(ClinicContext context)
+        {
+            _context = context;
+        }
+        public Task<Specialization> AddSpecializationAsync(Specialization specialization)
```
Comment (YuraRov) [inline DataAccess/Repositories/SpecializationRepository.cs:19]:
```suggestion
        public async Task<Specialization> AddSpecializationAsync(Specialization specialization)
```
The same thing for other async methods

**Review by YuraRov — CHANGES_REQUESTED** [review]

**Review by YuraRov — APPROVED** [review]

**Review by FedorivRostik — CHANGES_REQUESTED** [review]
issued extra usings

**Review by FedorivRostik — APPROVED** [review]

**Review by YuraRov — APPROVED** [review]

**Pull request merged**